File: C solutions/Prog_4.c
/*Given an array which consists of only 0, 1 and 2. Sort the array without using any sorting algo*/

#include<stdio.h>

int main()
{
    int n,i,p0,p1,p2;
    printf("Enter the size of the array: ");
    scanf("%d",&n);

    int arr[n];
    printf("\nEnter %d elements in the array only 0,1,2:\n",n);
    for(i= 0; i< n; i++)
    {
        do
        {
            scanf("%d",&arr[i]);
            if(arr[i] == 0 || arr[i]== 1 || arr[i]== 2)
                break;
            else 
            prinf("Enter only 0,1 or 2\n");
        } while (1);
        
        
        
    }
    
    
}

File: Java solutions/Prog4.java
/*
Given an array which consists of only 0, 1 and 2. Sort the array without using any sorting algo
 */

import java.util.Arrays;
public class Prog4 {
    public static void sortArray(int[] arr) {
        int low = 0; // pointer for 0
        int mid = 0; // pointer for 1
        int high = arr.length - 1; // pointer for 2

        while (mid <= high) {
            if (arr[mid] == 0) {
                // Swap arr[mid] and arr[low]
                int temp = arr[mid];
                arr[mid] = arr[low];
                arr[low] = temp;

                low++;
                mid++;
            } else if (arr[mid] == 1) {
                // Element is already at the correct position
                mid++;
            } else if (arr[mid] == 2) {
                // Swap arr[mid] and arr[high]
                int temp = arr[mid];
                arr[mid] = arr[high];
                arr[high] = temp;

                high--;
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = {2, 0, 1, 1, 0, 2};
        System.out.println("Original array: " + Arrays.toString(arr));

        sortArray(arr);
        System.out.println("Sorted array: " + Arrays.toString(arr));
    }
}

/*  Output
Original array: [2, 0, 1, 1, 0, 2]
Sorted array: [0, 0, 1, 1, 2, 2]
 */